tools/src/main/kotlin/ktx/tools/BundleLinesCreator.kt

```diff
@@ -0,0 +1,257 @@
+   * @param pathPrefix A prefix applied for all file operations. Default is empty, so the working directory is the
+   * parent.
+   */
+  fun execute(
+    targetPackage: String,
+    explicitBundlesDirectory: String? = null,
+    searchSubdirectories: Boolean = true,
+    targetSourceDirectory: String = "core/src",
+    enumClassName: String? = "Nls",
+    codeIndent: String = "    ",
+    pathPrefix: String = ""
+  ) {
+    require(codeIndent.all(Char::isWhitespace)) { "codeIntent must consist entirely of whitespace." }
+    val parentDirectory = explicitBundlesDirectory?.let { File(pathPrefix + it) }
+      ?: findFallbackAssetDirectory(pathPrefix)
+    requireNotNull(parentDirectory) { "Failed to find asset directory." }
+    val baseFiles = findBasePropertiesFiles(parentDirectory, searchSubdirectories)
+    val enumNamesToEntryNames = collectEnums(baseFiles, enumClassName)
+    val outDir = File("$pathPrefix$targetSourceDirectory/${targetPackage.replace(".", "/")}")
+      .apply { mkdirs() }
+    for ((enumName, entryNames) in enumNamesToEntryNames) {
+      val outFile = File(outDir, "$enumName.kt")
+      val sourceCode = generateKtFileContent(targetPackage, enumName, entryNames, codeIndent)
+      outFile.writeText(sourceCode)
+    }
+    println(
+      "Created BundleLine enum class(es) for bundles in directory \"$parentDirectory\":\n" +
+        enumNamesToEntryNames.keys.joinToString(separator = ",\n") { "  $it" } +
+        "\nin package $targetPackage in source directory \"$pathPrefix$targetSourceDirectory\"."
+    )
+  }
+
+  /**
+   * Finds the assets directory to be searched for properties files if no explicit directory was provided. The default
+   * implementation finds the first existing directory in descending precedence of `android/assets/i18n`,
+   * `android/assets/nls`, `android/assets`, `core/assets/i18n`, `core/assets/nls`, and `core/assets`.
+   * @return The parent directory, or null if none exists.
+   */
+  protected open fun findFallbackAssetDirectory(pathPrefix: String): File? {
+    return listOf(
+      "android/assets/i18n",
+      "android/assets/nls",
+      "android/assets/",
+      "core/assets/i18n",
+      "core/assets/nls",
+      "core/assets"
+    )
+      .map { File(pathPrefix + it) }
+      .firstOrNull(File::isDirectory)
+  }
+
+  /**
+   * Collects properties files that will be processed. The default implementation finds all files with name suffix
+   * `.properties` and no underscore in the name.
+   * @param parentDirectory The directory that will be searched.
+   * @param searchSubdirectories Whether to include sub-directories of the parent directory in the search.
+   * @return A list of all applicable properties files found.
+   */
+  protected open fun findBasePropertiesFiles(parentDirectory: File, searchSubdirectories: Boolean): Collection<File> {
+    val noUnderscoresFilter = object : AbstractFileFilter() {
```

Comment:
I believe you don't need this library to achieve this on JVM, try using `java.nio.file.Files#walk`.

Reply:
Also, I'd be surprised if Gradle didn't have similar utilities.

Reply:
The ktx project as a whole is set to Java compatibility 1.6, right? Not sure if we want to have to set up this module separately to be able to use Java 8 APIs. AFAIK, using the Gradle utilities would eliminate the ability to run this as anything but a Gradle task, as the file walking methods rely on a Gradle Project instance.

Reply:
Set this module to 8 explicitly, it's OK - Gradle runs on 8+ either way.